### PR TITLE
Remove deprecated client concurrency methods

### PR DESCRIPTION
**Pull request by jlowin**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] add a changelog entry in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
The Core client refers to `task_tag_...` routes that have been deprecated for some time in Cloud and don't exist in Server. The UI is the preferred way to set these limits.

### CODE REVIEW

**Comment by jlowin** [comment]
Ah, I just realized we didn't mark these as deprecated in Core, which was an oversight, although they've been deprecated for close to 6 months. Perhaps I shouldn't remove them wholesale quite yet.

File: docs/orchestration/concepts/task-concurrency-limiting.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ Once you have tagged your various tasks and [registered your Flow(s)](flows.html
 
 ### UI
 
-To set Task tag concurrency limits from the UI, go to Prefect Cloud and navigate to Team Settings -> Task Concurrency Limits.
+To set task concurrency limits from the UI, go to Prefect Cloud and navigate to Team Settings -> Task Concurrency Limits.
```
Comment (cicdw) [inline docs/orchestration/concepts/task-concurrency-limiting.md:42]: I don't see the benefit in removing the 'tag' descriptor here and elsewhere - the limits are based on tags

File: src/prefect/client/client.py [hunk 2]
```diff
@@ -1351,82 +1351,6 @@ def set_secret(self, name: str, value: Any) -> None:
         if not result.data.set_secret.success:
             raise ValueError("Setting secret failed.")
 
-    def get_task_tag_limit(self, tag: str) -> Optional[int]:
```
Comment (cicdw) [inline src/prefect/client/client.py:1354]: Why not update the GraphQL API routes that are being called and leave the client methods with the same name / signature?

**Comment by madkinsz** [comment]
Note that this PR is now stale with regards to the docs at https://docs.prefect.io/orchestration/flow-runs/concurrency-limits.html#task-run-limits but the deprecation still needs to be done on the core/client side

**Pull request closed without merging**